### PR TITLE
Codechange: use more C++11 in script-related functions

### PR DESCRIPTION
**Pull request by TrueBrain**
## Motivation / Problem

When trying to read the AI code (emphasis on trying), I run into a lot of very hard to read statements that can be solved by using a bit of `auto` and `for (.. : ..)` constructions.

## Description

Well .. it uses C++11 now.

## Limitations

- Only did matches for `iter`  in `script`, `ai` and `game` folder.

### CODE REVIEW

**Comment by frosch123** [comment]
* No, there is no find with default value like python's dict.get().
* More const is better, but sometimes hurts readability. Make your own judgement whether you can still read the code.
* Looks like GetConfigList never returns nullptr, so the return type should be "const ScriptConfigItemList&".

File: src/script/script_info.cpp [hunk 1]
```diff
@@ -232,8 +232,8 @@ SQInteger ScriptInfo::AddLabels(HSQUIRRELVM vm)
 	ValidateString(setting_name);
 
 	ScriptConfigItem *config = nullptr;
-	for (ScriptConfigItemList::iterator it = this->config_list.begin(); it != this->config_list.end(); it++) {
-		if (strcmp((*it).name, setting_name) == 0) config = &(*it);
+	for (auto &item : this->config_list) {
```
Comment (LordAro) [inline src/script/script_info.cpp:235]: this one can be const

File: src/script/script_scanner.cpp [hunk 2]
```diff
@@ -273,16 +271,16 @@ static bool IsSameScript(const ContentInfo *ci, bool md5sum, ScriptInfo *info, S
...
+		if (IsSameScript(ci, md5sum, item.second, this->GetDirectory())) return true;
 	}
 	return false;
 }
 
 const char *ScriptScanner::FindMainScript(const ContentInfo *ci, bool md5sum)
 {
-	for (ScriptInfoList::iterator it = this->info_list.begin(); it != this->info_list.end(); it++) {
-		if (IsSameScript(ci, md5sum, (*it).second, this->GetDirectory())) return (*it).second->GetMainScript();
+	for (const auto &item: this->info_list) {
```
Comment (LordAro) [inline src/script/script_scanner.cpp]: missing space

**Review by LordAro — COMMENTED** [review]
There are `contains` functions (which could cover most of the `find` cases)... but not until C++20
https://en.cppreference.com/w/cpp/container/map/contains

Could write helpers I suppose...

File: src/script/script_info.cpp (hunk 1 above)
Comment (TrueBrain) [reply src/script/script_info.cpp:235]: Compiler says no:

`script_info.cpp:236:54: error: invalid conversion from ‘const ScriptConfigItem*’ to ‘ScriptConfigItem*’ [-fpermissive]
  236 |   if (strcmp(item.name, setting_name) == 0) config = &item;`

(I tried `const` first for all, if fail, removed `const`)

Comment (LordAro) [reply src/script/script_info.cpp:235]: Ah yes, because `config` is not const

**Review by LordAro — APPROVED** [review]
My issues fixed

**Pull request merged**